WIP schema changes

zachlipton opened the pull request.

zachlipton committed:
Update init_test_data.sql

zachlipton committed:
use timestamptz

zachlipton committed:
Update init_test_data.sql

zachlipton committed:
Update init_test_data.sql

zachlipton committed:
Update init_test_data.sql

julia326 commented on init_test_data.sql:
```diff
@@ -2,31 +2,38 @@ USE data;
-    push_time DATETIME,
+    batch_id serial primary key,
+    push_time timestamptz not null,
+    publish_time timestamptz,
```
what's the difference between push_time and publish_time?

zachlipton replied:
based on Josh's work, I was thinking it could make sense to have push_time be the time the batch was pushed/created and publish_time be the time the batch actually got published. Maybe we can call them `created_at` and `published_at` for clarity?

julia326 commented on init_test_data.sql:
```diff
@@ -2,31 +2,38 @@ USE data;
+    is_daily_commit BOOLEAN not null,
+    is_preview BOOLEAN not null,
+    is_revision BOOLEAN not null
     PRIMARY KEY (batch_id)
```
can probably remove this line

julia326 commented on init_test_data.sql:
```diff
@@ -2,31 +2,38 @@ USE data;
 
+CREATE TABLE geography (
+    geography_id serial primary key,
+    abreviation char(2),
```
nit: "abbreviation" :)

julia326 commented on init_test_data.sql:
```diff
@@ -2,31 +2,38 @@ USE data;
 -- Create initial tables
 
+CREATE TABLE geography (
+    geography_id serial primary key,
```
thought on this. for readability, should we have the abbreviation be the PK so we don't need to join against this table for any visual scan of core_data? and is there any reason not to?

zachlipton replied:
I went back and forth on that repeatedly since last night. Making the abbreviation the PK makes things so much more human readable, but starts to fail if we want to start tracking other non-state geographies in the future. On balance though, I think I'm persuaded that readability is more important than something we may never need (being able to scroll through the data table without memorizing state ids sounds important for our sanity).

zachlipton committed:
spell abbreviation correctly

zachlipton committed:
Changes from review

Just use the state abbreviation for human readability and rename the geography table to state. We can deal with any other geography types in the future if needed.

Rename batch timestamp columns

Rename commit_note to batch_note for consistent terminology
